rootex/framework/components/visual/camera_visual_component.cpp

```diff
@@ -48,13 +102,73 @@ void CameraVisualComponent::postRender()
 {
 }
 
-void CameraVisualComponent::setPosition(Vector3 position)
+const Matrix& CameraVisualComponent::getView()
 {
-	m_Position = position;
-	m_ViewMatrix = Matrix::CreateLookAt(m_Position, m_Position + m_Direction, { 0.0f, 1.0f, 0.0f });
+	if (m_TransformComponent)
+	{
+		Vector3 position = m_TransformComponent->getPosition();
+		Matrix transform = m_TransformComponent->getAbsoluteTransform();
+		Quaternion rotation = m_TransformComponent->getRotation();
+		Vector3 direction = transform.Forward();
+		Vector3 up = transform.Up();
+		if (rotation.x != 0 || rotation.y != 0 || rotation.z != 0)
+		{
+			direction = XMVector3Rotate(direction, rotation);
+			up = XMVector3Rotate(up, rotation);
+		}
+		m_ViewMatrix = Matrix::CreateLookAt(position, position + direction, up);
+	}
+	return m_ViewMatrix;
 }
 
 void CameraVisualComponent::setViewTransform(const Matrix& view)
 {
 	m_ViewMatrix = view;
 }
+
+void CameraVisualComponent::setInactive()
+{
+	m_Active = false;
+}
+
+JSON::json CameraVisualComponent::getJSON() const
+{
+	JSON::json j;
+
+	j["aspectRatio"]["x"] = m_AspectRatio.x;
+	j["aspectRatio"]["y"] = m_AspectRatio.y;
+
+	return j;
+}
+
+#ifdef ROOTEX_EDITOR
+#include "imgui.h"
+#include "imgui_stdlib.h"
+void CameraVisualComponent::draw()
+{
+	ImGui::DragFloat2("##A", &m_AspectRatio.x, s_EditorDecimalSpeed);
+	ImGui::SameLine();
+	if (ImGui::Button("Aspect Ratio"))
+	{
+		m_AspectRatio = { 16.0f, 9.0f };
+	}
+	m_ProjectionMatrix = Matrix::CreatePerspective(1.0f, 1.0f * m_AspectRatio.y / m_AspectRatio.x, 0.5f, 100.0f);
```

Comment:
Don't do this every frame. Do this only when DragFloat2 returns true i.e. it is edited by the user